src/interaction/selection.ts

```diff
@@ -16,27 +16,38 @@ export function start(): void {
 }
 
 function recompute(): void {
-    let s = document.getSelection();
+    let current = document.getSelection();
 
-    if (selection !== null && data.start !== null && data.start !== getId(s.anchorNode)) {
+    // Bail out if we don't have a valid selection
+    if (current === null) { return; }
```

Comment:
If current === null and previous != null then should we send an event?

Reply:
We will automatically send the previous based on the timeout that we scheduled earlier. So previous will go out automatically if there's no valid current within the look ahead period.

Reply:
Wasns't obvious. Can you add more comment about the logic? It's a bit opaque. now. 

---
In reply to: [334240742](https://github.com/microsoft/clarity-js/pull/151#discussion_r334240742) [](ancestors = 334240742)